pallets/lbp/src/lib.rs

```diff
@@ -808,6 +828,25 @@ impl<T: Config> Pallet<T> {
 			Error::<T>::InsufficientAssetBalance
 		);
 
+		let fee_asset = pool_data.assets.0;
+		let fee_amount = if assets.asset_in == fee_asset {
+			Self::calculate_fees(&pool_data, amount_in)?
+		} else {
+			Self::calculate_fees(&pool_data, amount_out)?
+		};
+
+		if fee_asset == assets.asset_in {
+			ensure!(
+				T::MultiCurrency::free_balance(assets.asset_in, &pool_id) >= amount_in + fee_amount,
+				Error::<T>::InsufficientAssetBalance
+			);
+		} else {
+			ensure!(
+				T::MultiCurrency::free_balance(assets.asset_in, who) >= amount_out + fee_amount,
+				Error::<T>::InsufficientAssetBalance
+			);
+		}
```

Comment:
Is this correct? I think it should be this way:

```suggestion
		let fee_amount = if assets.asset_in == fee_asset {
			Self::calculate_fees(&pool_data, amount_in)?
		} else {
			Self::calculate_fees(&pool_data, amount_out)?
		};

		if fee_asset == assets.asset_in {
			ensure!(
				T::MultiCurrency::free_balance(assets.asset_in, who) >= amount_in + fee_amount,
				Error::<T>::InsufficientAssetBalance
			);
		} else {
			ensure!(
				T::MultiCurrency::free_balance(assets.asset_in, &pool_id) >= amount_out - fee_amount,
				Error::<T>::InsufficientAssetBalance
			);
		}
```

Reply:
these checks are not correct/irelevant, removed for now